### PR TITLE
FEATURE: `references` and `backReferences` FlowQuery operations

### PR DESCRIPTION
**Pull request by bwaidelich**
## usage

    # get first reference instance
    ${q(node).references().get(0)}

    # get first reference name
    ${q(node).references().get(0).name}

    # get property of the first reference itself
    ${q(node).references("someReferenceName").referenceProperty("someReferenceProperty")}
    
    # get property of first referenced node
    ${q(node).references().nodes().property("someNodeProperty")}

    # the same API for back references
    ${q(node).backReferences("someReferenceName").referenceProperty("someReferenceProperty")}

Resolves: #4144

### CODE REVIEW

File: Neos.ContentRepository.NodeAccess/Classes/FlowQueryOperations/ReferencesOperation.php [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+ * To render the reference name of the first match:
+ *
+ *     $q{node).references().get(0).name.value}
+ *
+ * The "property" operation can work on these to access any property on the reference relation:
+ *
+ *     ${q(node).references("someReferenceName").property("somePropertyName")}
+ *
+ * @see BackReferencesOperation
+ * @see ContentSubgraphInterface::findBackReferences()
```
Comment (bwaidelich) [inline Neos.ContentRepository.NodeAccess/Classes/FlowQueryOperations/ReferencesOperation.php]:
```suggestion
 * @see ContentSubgraphInterface::findBackReferences()
 * @api Used in FlowQuery
```

**Comment by bwaidelich** [comment]
## TODO

- [x] dedicated `referenceProperty` operation (revert `property` operation changes)
- [x] new `nodes` operation

**Comment by skurfuerst** [comment]
@bwaidelich don't we still need the `nodes()` FlowQuery operation? otherwise looks good :)

**Review by skurfuerst — APPROVED** [review]

**Comment by bwaidelich** [comment]
@skurfuerst doh, you're right. I'll take care!

**Pull request merged**